lms/services/oauth2_token.py

```diff
@@ -51,19 +52,23 @@ def get(self):
 
         :raise OAuth2TokenError: if we don't have an OAuth 2 token for the user
         """
-        try:
-            return (
-                self._db.query(OAuth2Token)
-                .filter_by(
-                    application_instance=self._application_instance,
-                    user_id=self._user_id,
-                )
-                .one()
+        token = (
+            self._db.query(OAuth2Token)
+            .join(ApplicationInstance)
+            .filter(
+                # We don't query by application_instance but any token belonging to an AI with the same GUID
```

Comment:
We do the same over the file service:

https://github.com/hypothesis/lms/blob/main/lms/services/file.py#L74